### ISSUE TITLE
Authorize a google developer account without committing a json or other credential info

### ISSUE THREAD

**Issue by masonnl** [google/google-api-ruby-client#370]
Hello @sqrrrl and any other maintainers.

I recently took interest in this project and tried to roll it into our app. One drawback is the lack of documentation on authenticating requests to the api. The [Ruby Google Auth library](https://github.com/google/google-auth-library-ruby#example-web) requires a path to a `google_credentials.json` stored in a specific environment variable. We eventually got it working without an explicitly linked credentials file, but it required a lot of code spelunky-ing.

Our solution instead of committing a json file with our Google Application credentials, were environment variables that essentially provide the same information this api client requires to authenticate. This way we can keep secure credentials outside of our repo and instead rely on the server. The three essential environment variables that we set are as follows:

``` sh
GOOGLE_ACCOUNT_TYPE="YOUR ACCOUNT TYPE" # aka: 'service'
GOOGLE_CLIENT_EMAIL="YOUR GOOGLE DEVELOPER EMAIL"
GOOGLE_PRIVATE_KEY="YOUR GOOGLE DEVELOPER API KEY"
```

With the above environment variables we were able to authorize our service app and make queries to the depths of Google. My only concern is that this is more of a workaround for the Google Auth library... Let me know what you think!

**Proposition:**

Provide an additional authorization process as described in the steps above to the README.md.

**Comment by sqrrrl** [comment]
It's not a workaround :) The use of ENV variables like tat is supported by the google auth library. Agree that docs could be improved here though. It's hard to discover that without looking at the code....

**Comment by neycodes** [comment]
Can this be explained in detail? It is still quite unclear how to authorize with ENV variables.

**Comment by masonnl** [comment]
@neycodes Sorry it is unclear, and apologies ahead of time, it has been a long time since I looked at this code. From what I remember, our team did not want to check-in any files with our google credentials to authenticate with Google's API (as shown in [Google Auth's example to authenticate](https://github.com/google/google-auth-library-ruby#example-web)). Our repo's standard at the time required that all 3rd party services our app made calls to (Google) should have their credentials stored at the server level. I find this to be clearly described by the team at Heroku, [**here**](https://devcenter.heroku.com/articles/config-vars). I suggest following the authentication example from the first link above, but instead of reading your credentials from a file, set some variables from your shell named as described in this PR and set their values specific to your google account's credentials (`GOOGLE_ACCOUNT_TYPE`, `GOOGLE_CLIENT_EMAIL`, and `GOOGLE_PRIVATE_KEY`.

**TL;DR:** Storing these three credentials as environment variables on the server is an alternative to storing them in a file and having your app open a file.

**Comment by neycodes** [comment]
@masonnl Thanks for responding.

I would prefer not to check-in the google credential file as well. I'm still a little shell-shocked that google would actually choose this means of authentication.

Now, I've set the ENV variables you mentioned but google auth does not "see" them, which is why I asked earlier for a little more detail. Unless i'm missing something, it looks like it doesn't work anymore.

**Comment by masonnl** [comment]
@neycodes, I dug back into the [Google Auth Library](https://github.com/google/google-auth-library-ruby) and found fewer results for the keys I've mentioned in this PR. It seems things have changed. I searched for any mention of "environment variable" in their issues and came up with an open [issue here](https://github.com/google/google-auth-library-ruby/issues/65). I believe that is likely where the conversation should continue since authentication is the source of concern. Sorry I can't be of more help. All that said, this PR should probably be effectively reverted. I'll leave that up to maintainers to make a decision on, @sqrrrl.

**Comment by neycodes** [comment]
@masonnl Thanks again for trying. It does seem that there is no current method to authenticate service accounts with the API ruby client at the moment.

**Comment by sqrrrl** [comment]
@neycodes -- googleauth supports service accounts, and via ENV too.

The variables mentioned by @masonnl are correct. If you want to use environment variables to supply the credentials, you have to set two keys:

```
GOOGLE_PRIVATE_KEY
CLIENT_EMAIL_VAR
```

See https://github.com/google/google-auth-library-ruby/blob/37ed189b2e5165243918fcde127ebe323e9a06b5/lib/googleauth/credentials_loader.rb#L79

You then get the credentials using `Google::Auth.get_application_default()`

Note that the loader has a set of priorities -- if `GOOGLE_APPLICATION_CREDENTIALS` is set it'll try to read from that instead.

If you continue to have issues, please share some code + some debug info (e.g. log the environment variables, any exceptions raised, etc..)

**Comment by deepakmahakale** [comment]
In order for the following to work please assign the following ENV vars
```
export GOOGLE_ACCOUNT_TYPE=service_account
export GOOGLE_CLIENT_ID=1234567890
export GOOGLE_PRIVATE_KEY="-----BEGIN PRIVATE KEY-----\nabc....xyz\n-----END PRIVATE KEY-----"
export GOOGLE_CLIENT_EMAIL="analytics@xyz.iam.gserviceaccount.com"
```
And then

```ruby
require 'google/apis/analyticsreporting_v4'

reporting_service = Google::Apis::AnalyticsreportingV4::AnalyticsReportingService.new

credentials = Google::Auth::ServiceAccountCredentials.make_creds(
  scope: 'https://www.googleapis.com/auth/analytics.readonly'
)

reporting_service.authorization = credentials
```

Same works fro google drive API as well 

Reference: https://github.com/googleapis/google-auth-library-ruby/tree/584ad57d7d72d9ffa395daa1dde4c48e04ab3c99#example-environment-variables

**Comment by MrT3313** [comment]
+1 for clarification on getting credentials with ENV vars instead of JSON file w/ [google-api-python-client](https://github.com/googleapis/google-api-python-client)

**Comment by sebieire** [comment]
There is way in Python to do it and I've just posted it on stackoverflow. You basically use the from_service_account_info method instead. See explanation here: [https://stackoverflow.com/a/70294018/11829999]
